pkg/imgpkg/cmd/imgpkg.go

```diff
@@ -69,6 +70,24 @@ func NewImgpkgCmd(o *ImgpkgOptions) *cobra.Command {
 	// This configurations forces all nodes to do not accept extra args, but the completion requires 1 extra arg
 	cmd.AddCommand(NewCompletionCmd())
 
+	err := cmd.LocalFlags().MarkHidden("azure-container-registry-config")
+	if err != nil {
+		//TODO: wrap around a useful err msg
+		panic(err.Error())
+	}
+
+	cmd.PersistentPreRunE = func(subCmd *cobra.Command, args []string) error {
+		registryAzureContainerConfigFlag := subCmd.Flag("registry-azure-container-registry-config")
+		if registryAzureContainerConfigFlag == nil {
+			return nil
+		}
+
+		if registryAzureContainerConfigFlag.Value.String() != "" {
+			return cmd.Flags().Set("azure-container-registry-config", registryAzureContainerConfigFlag.Value.String())
+		}
+		return cmd.Flags().Set("azure-container-registry-config", os.Getenv("IMGPKG_REGISTRY_AZURE_CONFIG"))
+	}
```

Comment:
should we move this whole thing into RegistryFlags. it has access to cmd and this code would be closer where we define "our" flag.